Add proposal Steering Committee application, terms, and role

joshuagl opened the pull request:
This proposal clarifies the SLSA Steering Committee, specifically with regards to:

* responsibilities
* difference to core maintainer responsibilities
* application, nomination and terms

joshuagl committed:
Add proposal Steering Commitee application, terms, and role

This proposal clarifies the SLSA Steering Commitee, specifically with
regards to:

* responsibilities
* difference to core maintainer responsibilities
* application, nomination and terms

Signed-off-by: Joshua Lock <joshuagloe@gmail.com>

MarkLodato commented on 0007/README.md:
```diff
@@ -0,0 +1,220 @@
+    project strategy, and serve as an escalation point in the event of a
+    disagreement.
+    *   These duties are much less frequent than those of a Maintainer.
+    *   We generally **expect an SC Member to also be a Core Maintainer**, or a
+        Subproject Maintainer for a critical subproject, in order to ensure
+        that SC Members are well versed in the project.
+    *   Some Core Maintainers might not be SC Members. This may be a stepping
+        stone to becoming an SC Member.
```
Suggestion from @brunodom was to soften this language. Maybe just say that:
- We expect SC Members to be well versed in the project (or to show expertise, or something like that)
- In practice SC Members are likely also Core Maintainers, but this is not a requirement.
- Remove the sentence about being a stepping stone.

joshuagl replied:
[aff2902](https://github.com/slsa-framework/slsa-proposals/pull/15/commits/aff29026d276b7dbb8659f75224ecf80cc3203ae) softens the language, as suggested.

@brunodom @MarkLodato PTAL.

brunodom replied:
LGTM.

MarkLodato commented on 0007/README.md:
```diff
@@ -0,0 +1,220 @@
+
+## Problem statement
+
+The original SLSA Steering Committee has passed its two year term and, per
```
```suggestion
The original SLSA Steering Committee (SC) has passed its two year term and, per
```

So make it clear what "SC" means in this doc.

joshuagl committed:
Address review feedback

Most notably, soften the language to be more open to SC Members who are not
core maintainers.

Signed-off-by: Joshua Lock <joshuagloe@gmail.com>

kimsterv commented:
+1 Lgtm!

MarkLodato reviewed (approved):
@brunodom and @kimsterv could you approve via the GitHub interface (Files Changed -> Review changes -> Approve) so that others can easily see the checkmark in the upper right? Thanks!

melba-lopez commented on 0007/README.md:
```diff
@@ -0,0 +1,221 @@
+*   SC **no longer has 7 fixed seats**, but instead has a minimum of 3 and
+    ideally about 5.
+    *   Same guidelines for Maintainers, without the hard minimum.
+*   Individuals **self-nominate** to become a Maintainer or SC Member by
```
where do they submit their nomination? and what are the timelines so people know ahead of time.  (i.e. vacations/life happens)

joshuagl replied:
Good questions. I expect we will figure thesse details out as part of the implementation (a PR against [slsa-framework/governance](https://github.com/slsa-framework/governance/)), but my initial suggestion would be:
* nominations are GitHub Issues
* timeline/cadence would be documented in the governance repository and the nomination period would be broadly communicated (in Slack, email, meetings, etc)

lehors replied:
I think it'd be better to have nominations made in the form of a PR that changes the list of maintainers or SC members as applicable. The PR description can be used to provide whatever information is required and this enforces the maintenance of the list in a publicly visible manner.

david-a-wheeler replied:
You really want an *odd* number when voting, or you risk deadlocks.

melba-lopez commented on 0007/README.md:
```diff
@@ -0,0 +1,221 @@
+    *   **Statement of intent** to continue performing duties over the next
+        year.
+    *   **Affiliation** / employer.
+*   **SC Members decide** if the individual meets the desirable properties
```
Shouldn't we allow the community to vote as well for the SC Members or maintainers seat?  SC can have the majority vote, but if you don't have community consensus, this leads to potential bias.

joshuagl replied:
We have been trying to avoid a formal voting process, as they can be onerous for the people running them. That also helps us avoid the question of how to identify elligible voters.

I hope that the public nomination process will help keep people honest. If we are to use GitHub Issues for nominations, that would allow folks to express support (though admittedly in a non-anonymous way).

MarkLodato replied:
IMO voting is overkill for such a small community. If the community grows 10x in size, then maybe we reconsider?

lehors replied:
The biggest challenge in holding such a vote is that it raises the question of who gets to vote. For this reason, I think it is better to keep the decision to the current maintainers and SC members even though it is prone to conflicts of interest.

david-a-wheeler replied:
I think it's important to have a mechanism for making hard decisions (voting), and then do your best to avoid using it (e.g., by determining why there are differences and working out compromises that address all major concerns).

melba-lopez commented on 0007/README.md:
```diff
@@ -0,0 +1,221 @@
+            50% of the existing SC Members, excluding the candidate.
+        *   Maintainer applications require explicit approval from at least 2
+            existing SC members (possibly including the candidate).
+*   Existing SC Members and Maintainers **annually re-apply** for the position,
```
When is the actual timeline?  every year on October... as an example.

abacchi replied:
For maintiners maybe it should be an open door period rather than one specific timeframe. SC member applicaiton should be a fixed annual application timeline though.

joshuagl replied:
Actual timeline tbd. I believe we should test the proposed change to the governance repo before we fully accept it, by running a nomination cycle for the SC. Then will do things annually from that (approximate) date.

Agreed that for maintainers, it should not be a specific timeframe. Good point!

MarkLodato replied:
My feeling was rolling, as @abacchi said. We could have some sort of script that sends a notification when a SC member or maintainer's approval is out of date.

melba-lopez commented on 0007/README.md:
```diff
@@ -0,0 +1,221 @@
+*   State explicitly that SLSA lives within and takes broad direction from the
+    OpenSSF SCI WG.
+
+## Alternatives considered
```
thanks for adding this.

melba-lopez commented on 0007/README.md:
```diff
@@ -0,0 +1,221 @@
+*   Set up Maintainers teams and add Maintainers as members if needed
+*   Change the org member base permissions to “No permissions” (from “Admin”)
+    and check that everyone still has necessary access
+*   Purge unnecessary permissions in the org and in each repo
```
may want to add a cadence.. quarterly.. every 6 months..etc.

joshuagl replied:
Good call, I'll add a note that we should do this regularly and at a minimum following each SC nomination cycle.

joshuagl replied:
Added in https://github.com/slsa-framework/slsa-proposals/pull/15/commits/4d91b1ab5415d562e4c3560d3edbbba923070d79

joshuagl committed:
Link to newly filed issue

Signed-off-by: Joshua Lock <joshuagloe@gmail.com>

abacchi commented on 0007/README.md:
```diff
@@ -0,0 +1,221 @@
+        and the overall project issue tracker.
+    *   A **Subproject Maintainer** is a maintainer of a different repo within
+        the slsa-framework organization.
+*   SC Member: a **senior leader** of the SLSA project who is **trusted** and
```
It may be nice for the SC group to present/review their activities quarterly / 6 month'ly at the standard SLSA meetings to improve transparency as well.

joshuagl replied:
I like the suggestion and agree with the principal. I am wary of codifying a periodicity, but added some text in [6a96a8c](https://github.com/slsa-framework/slsa-proposals/pull/15/commits/6a96a8cc37d3fe4d33321cacd63d49a58d6aae5e). WDYT?

abacchi replied:
yes that update looks great.

david-a-wheeler replied:
We want at least 2 different organizations represented, yes? Shouldn't that be clearly stated?

joshuagl committed:
Add note on regularly purging permissions

Signed-off-by: Joshua Lock <joshuagloe@gmail.com>

joshuagl committed:
Add SC Member communication responsibilities

Signed-off-by: Joshua Lock <joshuagloe@gmail.com>

joshuagl reviewed (commented):
Thanks all for the reviews and feedback so far. I've pushed some minor changes in response to review feedback:

* noted that we should perform purging of unneccessary GitHub permissions on a regular cadence
* added SC Member responsibilities:
   * sharing relevant SCI WG happeninings with the community (making this sharing 2-way)
   * that the SC should share updates in the community meeting (I've added a section to the meeting minute template in the Google doc too)

lehors commented on 0007/README.md:
```diff
@@ -0,0 +1,225 @@
+
+*   Status quo
+    *   see [goals]
+*   Dissolve the SLSA Steering Committee in favor of steering from the SCI WG
+    *   Con: Strategy may be driven by those without deep familiarity with
+        the project.
+    *   Con: SLSA community may not feel an appropriate sense of ownership and
+        autonomy.
```
For the sake of completeness this should include the pro:
```suggestion
*   Dissolve the SLSA Steering Committee in favor of steering from the SCI WG
    *   Pro: simplified governance structure
    *   Con: Strategy may be driven by those without deep familiarity with
        the project.
    *   Con: SLSA community may not feel an appropriate sense of ownership and
        autonomy.
```

joshuagl replied:
Agreed, added in [4432b4a](https://github.com/slsa-framework/slsa-proposals/pull/15/commits/4432b4a8dd41a8ee2ab73f1a074b409ab7e4e07e). Thanks!

lehors commented on 0007/README.md:
```diff
@@ -0,0 +1,225 @@
+        autonomy.
+*   Merge the Core Maintainer and Steering Committee Member roles
+    *   Pro: simplified structure
+        *   Counterpoint: the simplification does not result in too significant
+            a savings
```
Err... I think you're falling into subjectivity land on this one. :-)

joshuagl replied:
Fair comment, I've removed this in [4432b4a](https://github.com/slsa-framework/slsa-proposals/pull/15/commits/4432b4a8dd41a8ee2ab73f1a074b409ab7e4e07e).

lehors reviewed (commented):
Thanks for putting this together. I appreciate the candid approach about the current state we are in and think this is going in the right direction.
I added a few minor comments and have one nit:
The document uses both "OSSF" and "OpenSSF". I know OpenSSF is not consistent on this (we have openssf.org and github.com/ossf) but I think we should stick to OpenSSF which seems to be the dominant form.

brunodom commented:
@MarkLodato, unfortunately, I can't approve it now because I am traveling
and forgot my security key. I am not able to set up the Authenticator
without logging into my account. Basically, I am lockout these days.
However, consider it approved on my side.

Best Regards!
-Bruno Domingues

On Mon, Jul 24, 2023 at 10:05 AM Mark Lodato ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> @brunodom <https://github.com/brunodom> and @kimsterv
> <https://github.com/kimsterv> could you approve via the GitHub interface
> (Files Changed -> Review changes -> Approve) so that others can easily see
> the checkmark in the upper right? Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/slsa-framework/slsa-proposals/pull/15#pullrequestreview-1543446801>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABCTQCB7T36GCX7B2L2BO5TXRZXITANCNFSM6AAAAAA2SXDPSQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

joshuagl committed:
Be consistent with OpenSSF (not OSSF)

Signed-off-by: Joshua Lock <joshuagloe@gmail.com>

joshuagl committed:
Clear up Alternative pros/cons

- remove subjective counterpoint
- document implicit pro of dissolving the SC

Signed-off-by: Joshua Lock <joshuagloe@gmail.com>

bobcallaway commented:
LGTM as well

MarkLodato commented:
Friendly ping. What do we need to do to get this over the line and accepted?

lehors commented:
> Friendly ping. What do we need to do to get this over the line and accepted?

Who are you pinging and asking this to @MarkLodato ? For what it's worth, I approve. But I think if the current SC approves that should do it since technically the SC owns the governance from what I understand. If you want to be more open about it I suggest you raise it as a proposal to be approved on the next spec group call.

trishankatdatadog commented:
I would prefer that we get the entire current SC to approve this.

mlieberman85 commented:
I will double check and approve shortly. I've read it, but want to make sure I haven't missed anything.

joshuagl commented:
It would be good to have approval from the full, current, @slsa-framework/slsa-steering-committee for this proposal. 

- @MarkLodato, @kimsterv and @trishankatdatadog have approved the PR
- @brunodom [approved in a comment]9https://github.com/slsa-framework/slsa-proposals/pull/15#issuecomment-1667159496)
- we're still waiting for @david-a-wheeler and @mlieberman85

mlieberman85 reviewed (approved):
LGTM

david-a-wheeler commented on 0007/README.md:
```diff
@@ -0,0 +1,224 @@
+            permissions = “No permission”
+    *   Owner:
+        *   Used for SC Members, OpenSSF administrators, and (as needed)
+            Maintainers
```
I doubt you want that many people as owners. Owners can delete everything.

david-a-wheeler reviewed (approved):
It looks okay to me with the following issues:

1. We need to ensure it stays multi-organizational. I believe the current rule is "no more than 2 from 1 organization, just like OpenSSF"; that should continue.
2. Let's ensure there is no deadlock in the voting process. The simple approach is to have an odd number of people; another alternative is to have a tie-breaking rule. But if we have that much contention and *must* make a decision, then we need to have a way to break ties.

joshuagl commented:
Thanks for the review.

> It looks okay to me with the following issues:
> 
>     1. We need to ensure it stays multi-organizational. I believe the current rule is "no more than 2 from 1 organization, just like OpenSSF"; that should continue.

Completely agree. This requirement is currently part of our documented governance model and shall continue under this proposal.

>     2. Let's ensure there is no deadlock in the voting process. The simple approach is to have an odd number of people; another alternative is to have a tie-breaking rule. But if we have that much contention and _must_ make a decision, then we need to have a way to break ties.

Good call out. The current governance documents require up to 7 participants in the SC. This proposal changes that to "a minimum of 3 and ideally about 5". 
We should explicitly define a tie-breaking, as I think that is more resilient than trying to require an odd number of members of the SC.

joshuagl committed:
Capture that we need to define a tie-breaking mechanism for votes

Signed-off-by: Joshua Lock <joshuagloe@gmail.com>
